readme.md

```diff
@@ -15,834 +14,1113 @@
 > ### "Part of being a good steward to a successful project is realizing that writing code for yourself is a Bad Idea™. If thousands of people are using your code, then write your code for maximum clarity, not your personal preference of how to get clever within the spec."
 >_Idan_ _Gazit_
 
+## 1.2. Table of contents
+
+* [Code style guide](#2-code-style-guide)
+	* [Whitespace](#21-whitespace)
+	* [Beautiful Syntax](#22-beautiful-syntax)
+	* [Type Checking (Courtesy jQuery Core Style Guidelines)](#23-type-checking)
+	* [Conditional Evaluation](#24-conditional-evaluation)
+	* [Practical Style](#25-practical-style)
+	* [Naming](#26-naming)
+	* [Faces of `this`](#27-faces-of-this)
+	* [Misc](#28-misc)
+	* [Native & Host Objects](#29-native--host-objects)
+	* [Comments](#210-comments)
+* [Tests](#3-tests)
+	* [Manual tests](#31-manual-tests)
+
+# 2. Code style guide
+
+## 2.1 Whitespace
+- Always use tabs. Never use spaces for indentation (for both - code and comments).
+- Tab is 4 spaces.
```

Comment:
Overall, yes, but when you use e.g. vim, you can configure tab as 2 spaces. Although, I agree that it's no longer tab than, so it's redundant information (actually, misleading, because if you count tab in spaces, it's no longer tab).